tests/by-util/test_hashsum.rs

```diff
@@ -338,6 +365,30 @@ fn test_check_md5sum() {
     }
 }
 
+// GNU also supports one line sep
+#[test]
+fn test_check_md5sum_only_one_space() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+
+    for f in ["a", " b", "c"] {
+        at.write(f, &format!("{f}\n"));
+    }
+    at.write(
+        "check.md5sum",
+        "60b725f10c9c85c70d97880dfe8191b3 a\n\
+        bf35d7536c785cf06730d5a40301eba2  b\n\
+        2cd6ee2c70b0bde53fbe6cac3c8b8bb1 *c\n",
+    );
+    scene
+        .ccmd("md5sum")
+        .arg("--strict")
+        .arg("-c")
+        .arg("check.md5sum")
+        .succeeds()
+        .stdout_only("a: OK\n b: OK\nc: OK\n");
```

Comment:
GNU checks a different file here:

```console
$ ../gnu/src/md5sum --strict -c checksum
a: OK
../gnu/src/md5sum: ' b': No such file or directory
 b: FAILED open or read
../gnu/src/md5sum: '*c': No such file or directory
*c: FAILED open or read
../gnu/src/md5sum: WARNING: 2 listed files could not be read
[$? = 1]
$ cargo run -q md5sum --strict -c checksum
a: OK
md5sum:  b: No such file or directory
 b: FAILED open or read
c: OK
md5sum: WARNING: 1 listed file could not be read
[$? = 1]
```

This is a problem, as it does try to open the file named `*c`, and reads from it. This leaks information, and might cause a checksum verification to pass when it shouldn't.